shared/chat/conversation/input/mention-handlers.hoc.js

```diff
@@ -0,0 +1,243 @@
+// @flow
+import * as React from 'react'
+import {type PreMentionHocProps, type Props} from '.'
+import logger from '../../../logger'
+
+type MentionHocProps = {
+  channelMentionFilter: string,
+  channelMentionPopupOpen: boolean,
+  setChannelMentionFilter: (filter: string) => void,
+  setChannelMentionPopupOpen: (setOpen: boolean) => void,
+  mentionFilter: string,
+  mentionPopupOpen: boolean,
+  setMentionFilter: (filter: string) => void,
+  setMentionPopupOpen: (setOpen: boolean) => void,
+  _inputSetRef: any => void,
+} & PreMentionHocProps
+
+type MentionHocState = {
+  upArrowCounter: number,
+  downArrowCounter: number,
+  pickSelectedCounter: number,
+}
+
+const mentionHoc = (InputComponent: React.ComponentType<Props>) => {
```

Comment:
we want `Props & MentionHocState` here too, right?

Reply:
`Props = PreMentionHocProps & MentionHocProps` (from `index.js.flow`), and `MentionHocProps` includes the counters that are in `MentionHocState`